guides/source/action_mailbox_basics.md

```diff
@@ -277,13 +294,59 @@ class ApplicationMailbox < ActionMailbox::Base
 end
 ```
 
-Then set up a mailbox:
+The regular expression matches the incoming email's `to` field. For example, the above will match any email sent to `save@` to a 'forwards' mailbox, which we will need to create.
+
+### Create a Mailbox
 
 ```bash
 # Generate new mailbox
 $ bin/rails generate mailbox forwards
 ```
 
+This creates `app/mailboxes/forwards_mailbox.rb`, with a `ForwardsMailbox` class and a `process` method.
+
+### Process Email
+
+When processing an `InboundEmail`, you can get the parsed version of the email as a [`Mail`](https://github.com/mikel/mail) object with `InboundEmail#mail`. You can also get the raw source directly using the `#source` method. With `Mail` object, you can access the relevant fields, such as `mail.to`, `mail.body.decoded`, etc.
+
+```bash
+$ mail
+#<Mail::Message:33780, Multipart: false, Headers: <Date: Wed, 31 Jan 2024 22:18:40 -0600>, <From: someone@hey.com>, <To: save@example.com>, <Message-ID: <65bb1ba066830_50303a70397e@Bhumis-MacBook-Pro.local.mail>>, <In-Reply-To: >, <Subject: Hello Action Mailbox>, <Mime-Version: 1.0>, <Content-Type: text/plain; charset=UTF-8>, <Content-Transfer-Encoding: 7bit>, <x-original-to: >>
+$ mail.to
+["save@example.com"]
+$ mail.from
+["someone@hey.com"]
+$ mail.date
+Wed, 31 Jan 2024 22:18:40 -0600
+$ mail.subject
+"Hello Action Mailbox"
+$ mail.body
+#<Mail::Body:0x00007fc74cbf46c0 @boundary=nil, @preamble=nil, @epilogue=nil, @charset="US-ASCII", @part_sort_order=["text/plain", "text/enriched", "text/html", "multipart/alternative"], @parts=[], @raw_source="This is the body of the email message.", @ascii_only=true, @encoding="7bit">
+$ mail.body.decoded
+"This is the body of the email message."
+```
+### Inbound Email Status
+
+Once an email has been routed to a matching mailbox and processed, Action Mailbox updates the email status stored in `action_mailbox_inbound_emails` table with one of the following values:
+
+- Pending: Just received by one of the ingress controllers and scheduled for routing.
+- Processing: During active processing, while a specific mailbox is running its `process` method.
+- Delivered: Successfully processed by the specific mailbox.
+- Failed: An exception was raised during the specific mailbox’s execution of the #process method.
+- Bounced: Rejected processing by the specific mailbox and bounced to sender.
```

Comment:
Perhaps we should use the actual statuses as code here, like in the paragraph below, to avoid potential confusion, and make a better connection between them?

e.g.:

```suggestion
- `pending`: Just received by one of the ingress controllers and scheduled for routing.
- `processing`: During active processing, while a specific mailbox is running its `process` method.
- `delivered`: Successfully processed by the specific mailbox.
- `failed`: An exception was raised during the specific mailbox’s execution of the #process method.
- `bounced`: Rejected processing by the specific mailbox and bounced to sender.
```

Reply:
Yeah I like that better!